third_party/amd/lib/TritonAMDGPUToLLVM/ConvertLayoutOpToLLVM/SharedToDotOperandHelper.cpp

```diff
@@ -100,15 +140,31 @@ llvm::SmallVector<Value> computeOffsetsAType(
   auto order = srcLayout.getOrder();
   llvm::SmallVector<Value> aOffsets(blockSize * numBlocks);
 
-  for (int block = 0; block < numBlocks; ++block) {
-    int blockNonKOffset = block * nonKDim * warpsPerBlock;
-    Value offAdjust = mul(i32_val(blockNonKOffset), strides[rank - 2]);
-    for (int i = 0; i < blockSize; ++i) {
+  if (!isSwizzlePatternFitsIntoBlock(srcLayout, 0, reps, elemsPerInstr,
+                                     warpsPerBlock)) {
+    for (int block = 0; block < numBlocks; ++block) {
+      int blockNonKOffset = block * nonKDim * warpsPerBlock;
+      for (int i = 0; i < blockSize; ++i) {
+        Value row = add(mapping[i][0], i32_val(blockNonKOffset));
+        Value col = mapping[i][1];
+        aOffsets[block * blockSize + i] =
+            computeOffset(rewriter, loc, row, col, smemObj, srcLayout);
```

Comment:
I don't think this change directly related to dot3d enabling.
Is it ok to merge it in this pr?